### PR TITLE
use wc_format_stock_quantity_for_display() to wrap stock quantity 

### LINKED ISSUE #12753: Possibility to add a "suffix" to stock quantity
Specifically, in the art to cart validation messages, I have a use case where it would be helpful to have the error message read "You only have 4oz" in stock. That number is derived from `$product->get_stock_quantity()` which can be filtered, but if I filter in a suffix string (like oz) then it breaks any calculations that use `get_stock_quantity()` such as `has_enough_stock()`. Is there a way to to distinguish between calculations and front-end string display, such as an additional context?

### PR DESCRIPTION
**Pull request by helgatheviking**
numbers for front end display. closes #12753.

### CODE REVIEW

**Comment by mikejolley** [comment]
Hmm still a possible problem with this @helgatheviking. Without product context, how can you be sure you're appending to the correct qty's?

File: includes/class-wc-cart.php [hunk 1]
```diff
@@ -490,7 +490,7 @@ public function check_cart_item_stock() {
 			 */
 			if ( ! $product->has_enough_stock( $product_qty_in_cart[ $product->get_stock_managed_by_id() ] ) ) {
 				/* translators: 1: product name 2: quantity in stock */
-				$error->add( 'out-of-stock', sprintf( __( 'Sorry, we do not have enough "%1$s" in stock to fulfill your order (%2$s in stock). Please edit your cart and try again. We apologise for any inconvenience caused.', 'woocommerce' ), $product->get_name(), $product->get_stock_quantity() ) );
+				$error->add( 'out-of-stock', sprintf( __( 'Sorry, we do not have enough "%1$s" in stock to fulfill your order (%2$s in stock). Please edit your cart and try again. We apologise for any inconvenience caused.', 'woocommerce' ), $product->get_name(), wc_format_stock_quantity_for_display($product->get_stock_quantity() ) ) );
```
Comment (mikejolley) [inline includes/class-wc-cart.php]: Missing space after bracket here.

**Comment by helgatheviking** [comment]
I'm not sure I am following you? Currently, the way I have it isn't very different from the other formatting functions, ex: weight and dimensions. Are you thinking the `$product` object should be passed to the function and thus the filter?

**Comment by mikejolley** [comment]
Ok, let me give you an example.

You have 1 product in your store that is weight based and hence you want to add `oz` after the qty. How do you target just that product and no others?

**Comment by helgatheviking** [comment]
Thanks for the example. We could do something like this:

```
/**
 * Format the stock quantity ready for display.
 * @since  2.7.0
 * @param  int  $stock_quantity
 * @param  obj WC_Product
 * @return string
 */
function wc_format_stock_quantity_for_display( $stock_quantity, $product ) {
	return apply_filters( 'woocommerce_format_stock_quantity', $stock_quantity, $product );
}
```

**Comment by mikejolley** [comment]
Made an adjustment based on that. Had to change the other display function too so it would pass the product object.

**Comment by helgatheviking** [comment]
Cool, thanks Mike.

**Pull request merged**